sw/device/lib/base/memory.c

```diff
@@ -12,26 +14,109 @@
 #define OT_PREFIX_IF_NOT_RV32(name) ot_##name
 #endif
 
+static size_t compute_num_leading_bytes(uintptr_t left, uintptr_t right,
+                                        size_t len) {
+#ifndef OT_PLATFORM_RV32
+  assert(len == 0 || left != (uintptr_t)NULL);
+#endif
```

Comment:
We don't use `assert()` in device code. I see the `ifndef` but it would be better if we could cover this with tests.

```suggestion
```